File: mobile/src/main/java/info/papdt/express/helper/services/FCMInstanceIdService.java
package info.papdt.express.helper.services;

import android.util.Log;
import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

public class FCMInstanceIdService extends FirebaseInstanceIdService {

	@Override
	public void onTokenRefresh() {
		// Get updated InstanceID token.
		String refreshedToken = FirebaseInstanceId.getInstance().getToken();
		Log.d(FCMInstanceIdService.class.getSimpleName(), "Refreshed token: " + refreshedToken);

		// sendRegistrationToServer(refreshedToken);
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/services/FCMInstanceIdService.kt
package info.papdt.express.helper.services

import android.util.Log
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import info.papdt.express.helper.api.PushApi
import info.papdt.express.helper.support.Settings
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class FCMInstanceIdService : FirebaseInstanceIdService() {

	override fun onTokenRefresh() {
		// Get updated InstanceID token.
		val refreshedToken = FirebaseInstanceId.getInstance().token
		Log.d(FCMInstanceIdService::class.java.simpleName, "Refreshed token: " + refreshedToken!!)

		Settings.getInstance(this).putString(Settings.KEY_FIREBASE_INSTANCE_ID, refreshedToken)
		CoroutineScope(Dispatchers.IO).launch { PushApi.register(refreshedToken) }
	}

}
